ml-agents/mlagents/trainers/behavior_id_utils.py

```diff
@@ -0,0 +1,32 @@
+from typing import Dict, NamedTuple
+
+
+class BehaviorIdentifiers(NamedTuple):
+    brain_name: str
+    behavior_ids: Dict[str, int]
+
+    @staticmethod
+    def from_name_behavior_id(name_behavior_id: str) -> "BehaviorIdentifiers":
+        """
+        Parses a name_behavior_id into a BehaviorIdentifiers NamedTuple.
+        This allows you to access the brain name and distinguishing identifiers
+        without parsing more than once.
+        :param name_behavior_id: String of behavior params in HTTP format.
+        :returns: A BehaviorIdentifiers object.
+        """
+
+        ids: Dict[str, int] = {}
+        if "?" in name_behavior_id:
+            name, identifiers = name_behavior_id.split("?")
```

Comment:
Can you add some unit tests for this? 

In particular, I think this will choke on names that have a "?" in them e.g. "brainName?whatsThat?team=0"
``` python
>>> name_behavior_id = "brainName?whatsThat?team=0"
>>> name_behavior_id.split("?")
['brainName', 'whatsThat', 'team=0']
>>> name_behavior_id.rsplit("?", 1)
['brainName?whatsThat', 'team=0']
```
`rsplit` will start splitting from the right, and the second arg will limit the number of splits to 1. (sorry for not thinking of this in the original PR)

Reply:
Should I just disallow the use of '?' in names on the C# side?

Reply:
Better to support them if you can - just changing to `.rsplit("?", 1)` should be enough

Reply:
fixed